app/controllers/invoices_controller.rb

```diff
@@ -27,12 +30,31 @@ def create
    
     def show
         @invoice = Invoice.find(params[:id])
+        @client = @invoice
     end
     def edit
         @invoice = Invoice.find(params[:id])
     end
+    def update
+        @invoice = Invoice.find(params[:id])
+        #byebug
+         if @invoice.update_attributes(invoice_params)
+          redirect_to(:action => 'show', :id => @invoice.id) 
+        else 
+          render 'edit' 
+        end   
+    end
+    def destroy
+      @invoice = Invoice.find(params[:id])
+      @invoice.destroy
+      redirect_to invoices_path
+    end
+    def view_invoice
```

Comment:
By convention we usually put a blank line in-between methods. It might seem relatively minor but it makes it far easier for people to visualise and work with your code, so has a real benefit to building software in teams.